recipes/sdbus-cpp/all/conanfile.py

```diff
@@ -48,74 +42,98 @@ def _minimum_compilers_version(self):
         if self.options.with_code_gen:
-            self.build_requires("expat/2.4.1")
+            self.tool_requires("expat/2.4.8")
 
     def requirements(self):
-        self.requires("libsystemd/249.4")
+        self.requires("libsystemd/251.4")
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
 
     def source(self):
-        tools.get(**self.conan_data["sources"][self.version],
-            destination=self._source_subfolder, strip_root=True)
-
-    def _configure_cmake(self):
-        if self._cmake:
-            return self._cmake
-        self._cmake = CMake(self)
-        self._cmake.definitions["BUILD_CODE_GEN"] = self.options.with_code_gen
-        self._cmake.definitions["BUILD_DOC"] = False
-        self._cmake.definitions["BUILD_TESTS"] = False
-        self._cmake.definitions["BUILD_LIBSYSTEMD"] = False
-        self._cmake.configure()
-        return self._cmake
-
-    def _patch_sources(self):
-        for patch in self.conan_data.get("patches", {}).get(self.version, []):
-            tools.patch(**patch)
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.variables["BUILD_CODE_GEN"] = self.options.with_code_gen
+        tc.variables["BUILD_DOC"] = False
+        tc.variables["BUILD_TESTS"] = False
+        tc.variables["BUILD_LIBSYSTEMD"] = False
+        tc.generate()
+
+        # workaround for https://gitlab.kitware.com/cmake/cmake/-/issues/18150
+        copy(self, "*.pc", self.generators_folder,
+             os.path.join(self.generators_folder, "lib", "pkgconfig"))
 
     def build(self):
-        self._patch_sources()
-        cmake = self._configure_cmake()
+        apply_conandata_patches(self)
+        cmake = CMake(self)
+        cmake.configure()
         cmake.build()
 
     def package(self):
-        cmake = self._configure_cmake()
+        cmake = CMake(self)
+        cmake.configure()
         cmake.install()
-        self.copy(pattern="COPYING", dst="licenses", src=self._source_subfolder)
-        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
-        tools.rmdir(os.path.join(self.package_folder, "lib", "cmake"))
+        copy(self, "COPYING*", self.source_folder,
+             os.path.join(self.package_folder, "licenses"))
+        rmdir(self, os.path.join(self.package_folder, "lib", "pkgconfig"))
+        rmdir(self, os.path.join(self.package_folder, "lib", "cmake"))
 
     def package_info(self):
+        self.cpp_info.set_property("cmake_file_name", "sdbus-c++")
+        self.cpp_info.components["sdbus-c++"].libs = ["sdbus-c++"]
+        self.cpp_info.components["sdbus-c++"].requires.append(
+            "libsystemd::libsystemd")
+        self.cpp_info.components["sdbus-c++"].set_property(
+            "cmake_target_name", "SDBusCpp::sdbus-c++")
+        self.cpp_info.components["sdbus-c++"].set_property(
+            "pkg_config_name", "sdbus-c++")
```

Comment:
Also define `cmake_target_name` & `pkg_config_name` to avoid unofficial target `sdbus-cpp::sdbus-cpp` (problematic since it's the one promoted in `markdown` generator).